modules/order/src/Entity/OrderItem.php

```diff
@@ -108,6 +110,26 @@ public function getQuantity() {
   /**
    * {@inheritdoc}
    */
+  public function getItemsQuantity() {
+    $settings = $this->getQuantityWidgetSettings();
+    $step = isset($settings['#step']) && is_numeric($settings['#step']) ? $settings['#step'] + 0 : 1;
+    $quantity = $this->getQuantity();
+    return (string) is_int($step) ? $quantity : (is_float($step) && $quantity > 0 ? '1' : $quantity);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getQuantityWidgetSettings() {
```

Comment:
An entity should not be aware of, or care for, the widget settings of a field. They're two completely different API levels.